imsms_analysis/analysis_runner.py

```diff
@@ -73,6 +77,105 @@ def run(self, factory):
+                target, 
+                _, _
+                ) = run_preprocessing(config, self.callbacks)
+
+                base_dir = "dataset"
+                dataset = "imsms-mlab"
+                preparation = config.analysis_name
+                target_name = "disease_binary"
+                algorithm = config.mlab_algorithm
+                if algorithm is None:
+                    algorithm = "RandomForestClassifier"
+
+                # Create the  expected file structure
+                results_dir = path.join(
+                    base_dir, dataset, preparation, target_name
+                )
+                if not path.exists(results_dir):
+                    makedirs(results_dir)
+                
+                # Save table and metadata
+                table_fp = path.join(
+                    base_dir, dataset, preparation, target_name,
+                    "filtered_table.qza"
+                )
+                table_artifact = Artifact.import_data(
+                    "FeatureTable[Frequency]", 
+                    final_biom
+                )
+                table_artifact.save(table_fp)
+
+                metadata_fp = path.join(
+                    base_dir, dataset, preparation, target_name,
+                    "filtered_metadata.qza"
+                )
+                metadata_artifact = Artifact.import_data(
+                    "SampleData[Target]", target
+                )
+                metadata_artifact.save(metadata_fp)
+                
+
+                # run job indices 1...10 inclusive, letting 5 jobs run at once
+                # each job has 100 parameter sets, for a total of 1000 parameter sets
+                start = 1
+                end = 10
+                n_concurrent_jobs = 5
+                chunk_size = 100
+                (
+                    script_fp,
+                    params_fp,
+                    run_info_fp,
+                ) = orchestrate_hyperparameter_search(
+                    dataset=dataset,
+                    preparation=preparation,
+                    target=target_name,
+                    algorithm=algorithm,
+                    repeats=3,  # num CV repeats
+                    base_dir=base_dir,  # Directory with mlab structure containing datasets
+                    ppn=4,  # processors per node
+                    memory=32,  # memory in GB
+                    wall=50,  # walltime  in hours
+                    chunk_size=chunk_size,  # num parameter sets to run per job
+                    randomize=True,  # randomly shuffle order of parameter set
+                    force=False,  # force overwrite of existing results
+                    dry=False,  # dry runs
+                    dataset_path=table_fp,
+                    metadata_path=metadata_fp,
+                )
+                
+                cmd = ["qsub", "-t", f"{start}-{end}%{n_concurrent_jobs}", script_fp]
+                subprocess.run(cmd)
```

Comment:
What is the -t?  Is f"{start}-{end}%{n_concurrent_jobs}" supposed to be the name of the job?  If so, can we append the name in the config?  It looks like all configs will have the same arguments passed in, just different script_fp values.

Reply:
-t is a flag to qsub telling it this is an array job, and it accepts arguments in the form {start}-{end}%{n_concurrent_jobs} so `qsub -t 1:50%10 jobscript.sh` will launch an array job with indices 1...50 inclusive (first 50 chunks, which  each have `chunk_size` parameter sets), and only allow up to 10 of those jobs to run at once.

Reply:
The job name is given by the  template in `q2_mlab/templates/array_job_template.sh `and is made of the dataset, target, 'level' (which is normally 16S or MG, but here we use the analysis name as 'level'), and algorithm.

Reply:
Ahh, so if I say start=1, end=10, it will run 10 jobs within the array job, each of which will randomly choose chunk_size parameter sets to try out?   

It looks like this will start one job array per config then, rather than one large job array containing all configs.  That may make it harder to figure out when the job is complete and whether or not all configs processed correctly.

Reply:
> Ahh, so if I say start=1, end=10, it will run 10 jobs within the array job, each of which will randomly choose chunk_size parameter sets to try out?
yep, more exactly it's taking the first 10 chunks from the randomly-ordered parameter list. If chunk_size=20, the job 1 will be the first 20 parameter sets, job 2 the next 20, etc. 

And also yup, one job array per config. Each job array is a hyperparameter search for that config, split into chunks over several jobs.

Reply:
> That may make it harder to figure out when the job is complete and whether or not all configs processed correctly.

yeah...I'm working on something for that.